app/assets/javascripts/components/projects/ProjectSearch.jsx

```diff
@@ -1,124 +1,108 @@
-        },
-        unjoined: {
-          title: I18n.t('projects.not_member_of'),
-          projects: this.filterProjects(
-            this.props.projects.unjoined,
-            projectsSearch,
-            projectsFilter
-          ),
-          joined: false,
-        },
+  const handleSearch = (searchValue, filterValue) => {
+    setVisibleProjects({
+      joined: {
+        title: I18n.t('projects.mine'),
+        projects: filterProjects(projects.joined, searchValue, filterValue),
+        joined: true,
+      },
+      unjoined: {
+        title: I18n.t('projects.not_member_of'),
+        projects: filterProjects(projects.unjoined, searchValue, filterValue),
+        joined: false,
       },
     });
-  }
+  };
 
-  filterOptions() {
-    return Project.filters.map(filter => {
-      return (
-        <option key={filter} value={filter}>
-          {I18n.t(filter)}
-        </option>
-      );
-    });
-  }
+  const filterOptions = () => {
+    return Project.filters.map(filter => (
+      <option key={filter} value={filter}>
+        {I18n.t(filter)}
+      </option>
+    ));
+  };
 
-  renderProjectList(list) {
+  const renderProjectList = list => {
     if (list.projects.length > 0) {
       return (
         <ProjectList
           title={list.title}
           projects={list.projects}
-          user={this.state.user}
+          user={user}
           joined={list.joined}
         />
       );
     }
-  }
-
-  render() {
-    return (
-      <Fragment>
-        <div className="search-projects">
-          <div className="form-group col-md-12">
-            <div className="input-group">
-              <div className="input-group-addon">
-                <i className="mi md-20 heading-icon">search</i>
-              </div>
-              <input
-                id="projects_search"
-                className="form-control"
-                onChange={this.handleSearch}
-                ref="projectsSearch"
-                placeholder="Search projects"
-              />
+  };
 
-              <div className="input-group-addon">
-                <select
-                  id="project_type"
-                  className="unstyled-input"
-                  data-testid="select-project-filter"
-                  onChange={this.handleSearch}
-                  ref="projectsFilter"
-                >
-                  {this.filterOptions()}
-                </select>
-              </div>
+  return (
+    <Fragment>
+      <div className="search-projects">
+        <div className="form-group col-md-12">
+          <div className="input-group">
+            <div className="input-group-addon">
+              <i className="mi md-20 heading-icon">search</i>
+            </div>
+            <input
+              id="projects_search"
+              className="form-control"
+              onChange={e => {
+                setProjectsSearch(e.target.value);
+                handleSearch(e.target.value, projectsFilter);
+              }}
```

Comment:
What about creating functions for `handleProjectSearchChange` and `handleProjectFilterChange`?
```typescript
const handleProjectSearchChange = (e) => {
   setProjectsSearch(e.target.value);
   handleSearch(e.target.value, projectsFilter);
}

const handleProjectFilterChange = (e) => {
   setProjectsFilter(e.target.value);
   handleSearch(projectsSearch, e.target.value);
}